compysition/actors/httpserver.py

```diff
@@ -235,156 +235,168 @@ def __init__(self, name, address="0.0.0.0", port=8080, keyfile=None, certfile=No
-
-            local_response.body = response_data
-
+        if response_queue is not None:
+            event = self._process_response_accept(event=event, original_event_class=original_event_class)
+            local_response = self._create_response(event=event)
             response_queue.put(local_response)
             response_queue.put(StopIteration)
             self.logger.info("[{status}] Service '{service}' Returned in {time:0.0f} ms".format(
-                service=event.service,
-                status=local_response.status,
-                time=(datetime.now()-event.created).total_seconds() * 1000), event=event)
+                    service=event.service,
+                    status=local_response.status,
+                    time=(datetime.now()-event.created).total_seconds() * 1000),
+                event=event)
         else:
             self.logger.warning("Received event response for an unknown event ID. The request might have already received a response", event=event)
 
     def _format_bottle_env(self, environ):
         """**Filters incoming bottle environment of non-serializable objects, and adds useful shortcuts**"""
-
-        query_string_data = {}
-        for key in environ["bottle.request"].query.iterkeys():
-            query_string_data[key] = environ["bottle.request"].query.get(key)
-
-        environ = {key: environ[key] for key in environ.iterkeys() if isinstance(environ[key], (str, tuple, bool, dict))}
+        query_string_data = {key: value for key, value in environ["bottle.request"].query.iteritems()}
+        environ = {key: value for key, value in environ.iteritems() if isinstance(value, (str, tuple, bool, dict))}
         environ['QUERY_STRING_DATA'] = query_string_data
-
         return environ
 
-    def callback(self, queue=None, *args, **kwargs):
-        queue_name = queue or self.name
-        queue = self.pool.outbound.get(queue_name, None)
-        ctype = request.content_type.split(';')[0]
-
+    def _get_accept(self):
         accept_header = request.headers.get("Accept", "*/*")
-        try:
-            accept = mimeparse.best_match(self.CONTENT_TYPES, accept_header)
-        except ValueError:
-            accept = "*/*"
-            self.logger.warning("Invalid mimetype defined in client Accepts header. '{accept}' is not a valid mime type".format(accept=accept_header))
-
+        with ignore(ValueError):
+            return mimeparse.best_match(self.CONTENT_TYPES, accept_header)
+        self.logger.warning("Invalid mimetype defined in client Accepts header. '{accept}' is not a valid mime type".format(accept=accept_header))
+        '''
+        # What is the functionality here? Am I missing something? Perhaps this is incomplete.
         if request.method in ["GET", "OPTIONS", "HEAD", "DELETE"]:
             for accept_type in accept_header:
                 if self.CONTENT_TYPE_MAP.get(accept_type, None):
                     pass
+        '''
```

Comment:
I would like to draw attention to this as I need help understanding what this was intended for.

Reply:
That's a question we may never know the answer to; looks like it is just incomplete, I'd say it's fine to just remove it.